src/Mvc/test/Mvc.FunctionalTests/NonNullableReferenceTypesTest.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Reflection;
+using System.Threading.Tasks;
+using AngleSharp.Dom.Html;
+using AngleSharp.Parser.Html;
+using BasicWebSite.Models;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.FunctionalTests
+{
+    public class NonNullableReferenceTypesTest : IClassFixture<MvcTestFixture<BasicWebSite.StartupWithoutEndpointRouting>>
+    {
+        // Some tests require comparing the actual response body against an expected response baseline
+        // so they require a reference to the assembly on which the resources are located, in order to
+        // make the tests less verbose, we get a reference to the assembly with the resources and we
+        // use it on all the rest of the tests.
+        private static readonly Assembly _resourcesAssembly = typeof(BasicTests).GetTypeInfo().Assembly;
+
+        public NonNullableReferenceTypesTest(MvcTestFixture<BasicWebSite.StartupWithoutEndpointRouting> fixture)
+        {
+            Client = fixture.CreateDefaultClient();
+        }
+
+        private HttpClient Client { get; set; }
+
+        [Fact]
+        public async Task CanUseNonNullableReferenceType_WithController_OmitData_ValidationErrors()
+        {
+            // Arrange
+            var parser = new HtmlParser();
+
+            // Act 1
+            var response = await Client.GetAsync("http://localhost/NonNullable");
+
+            // Assert 1
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
```

Comment:
If you do `response.AssertStatusCodeAsync(HttpStatusCode.OK)`, it'll include the body if this is not OK. Way more useful than a `Expected: OK, Actual: InternalServerError`